draft-normington-jsonpath-latest.md

```diff
@@ -84,43 +108,266 @@ informative:
 
 normative:
   RFC3629:
-  RFC5234:
+  RFC5234: abnf
+  RFC8259: json
 
 --- abstract
 
-
 JSONPath defines a string syntax for identifying values
 within a JavaScript Object Notation (JSON) document.
 
 --- note_
 
 **This document is a work in progress and has not yet been published
-as an Internet Draft.**
+as an Internet Draft** (which needs to be fixed soon).
+
+--- note_Contributing
+
+This document picks up the popular JSONPath specification dated
+2007-02-21 and provides a more normative definition for it.
+In its current state, it is a strawman document showing what needs to
+be covered.
+
+Comments and issues can be directed at the github repository *insert
+repo here* as well as (for the time when the more permanent home is
+being decided) at the dispatch@ietf.org mailing list.
 
 --- middle
 
 # Introduction
 
-JSONPath was introduced by Stefan Goessner as a simple
-form of XPath for JSON.
-See his original article {{Goessner}}.
+This document picks up the popular JSONPath specification dated
+2007-02-21 {{JSONPath-orig}} and provides a more normative definition for it.
+In its current state, it is a strawman document showing what needs to
+be covered.
 
 JSON is defined by {{RFC8259}}.
 
-## Requirements Language
+JSONPath is not intended as a replacement, but as a more powerful
+companion, to JSON Pointer {{RFC6901}}. \[insert reference to section
+where the relationship is detailed.]
 
-{::boilerplate bcp14}
 
-## ABNF Syntax
+## Terminology
 
-The syntax in this document conforms to ABNF as defined by {{RFC5234}}.
+{::boilerplate bcp14}
 
+The grammatical rules in this document are to be interpreted as ABNF,
+as described in {{-abnf}}.
 ABNF terminal values in this document define Unicode code points rather than
 their UTF-8 encoding.
 For example, the Unicode PLACE OF INTEREST SIGN (U+2318) would be defined
 in ABNF as `%x2318`.
 
 
+The terminology of {{-json}} applies.
+
+Data Item:
+: A structure complying to the generic data model of JSON, i.e.,
+  composed of containers such as arrays and maps (JSON objects), and
+  of atomic data such as null, true, false, numbers, and text strings.
+
+Object:
+: Used in its generic sense, e.g., for programming language objects.
+  When a JSON Object as defined in {{-json}} is meant, we specifically
+  say JSON Object.
+
+Query:
+: Short name for JSONPath expression.
+
+Argument:
+: Short name for the JSON data item a JSONPath expression is applied to.
+
+Output Path:
```

Comment:
I'm in two minds. One of the main advantages of targeting the jsonpath specification at describing how an engine produces a sequence of _output paths_ as opposed to a sequence of _values_, is that it actually simplifies discussing about what a "value" actually is and removes potential causes of confusion that may arise in case an engine uses a compressed representation of a json tree, e.g. one where repeated subtrees may be actually reused.

Reply:
Opened #44 to discuss whether we need output for the initial draft.  I vote we leave it for now.